File: src/main/java/com/polygloat/dtos/request/ImportDto.java
package com.polygloat.dtos.request;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.util.Map;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ImportDto {
    @NotEmpty
    String languageAbbreviation;

    @NotNull
    Map<String, String> data;
}

File: src/main/kotlin/io/tolgee/dtos/ImportDto.kt
package io.tolgee.dtos

import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull

data class ImportDto(
        @field:NotEmpty
        val languageAbbreviation: String? = null,

        @field:NotNull
        var data: Map<String, String>? = null
)
